### PR TITLE
Control packit-service with the command embedded in a comment.

### LINKED ISSUE #412: enable controlling packit-service with the command embedded in a comment
Right now, the whole comment needs to match the command, so:
```
/packit build
```
But I'd like something like this:
```
I am going to reschedule builds using the newest deployment
/packit build
```
and not doing two comments

WDYT?

### PR DESCRIPTION
**Pull request by IceWreck**
Fixes #412 

It can now process commands embedded in comments:

- I forgot to whitelist it again. /packit build Hope this works.
- Last test. /packit build
- /packit build

The above examples have been tested and all of them work.

Screenshot_2020-02-27 Change Something Random by IceWreck · Pull Request #5 · IceWreck hello-world

### CODE REVIEW

**Comment by IceWreck** [comment]
Is the Zull build failing because of #432 ?

**Comment by TomasTomecek** [comment]
@IceWreck yes, but it got solved magically yesterday
recheck

File: packit_service/worker/jobs.py [hunk 1]
```diff
@@ -134,9 +134,11 @@ def process_comment_jobs(
         self, event: Union[PullRequestCommentEvent, IssueCommentEvent]
     ) -> HandlerResults:
         # packit_command can be `/packit propose-update`
-        msg = f"comment '{event.comment[:35]}'"
+        msg = f"comment '{event.comment[:100]}'"
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: we should take the whole comment, so please drop the `[:100]` part

File: packit_service/worker/jobs.py [hunk 2]
```diff
@@ -134,9 +134,11 @@ def process_comment_jobs(
...
         try:
-            (packit_mark, *packit_command) = event.comment.split(maxsplit=3)
+            cmd_start_index = event.comment.find(REQUESTED_PULL_REQUEST_COMMENT) 
+            (packit_mark, *packit_command) = event.comment[cmd_start_index:].split(maxsplit=3)
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: this is probably good enough for now, but once the commands start accept options and arguments, we'd need to improve heuristics

File: packit_service/worker/jobs.py (hunk 1 above)
Comment (IceWreck) [reply packit_service/worker/jobs.py]: Fixed.

**Comment by TomasTomecek** [comment]
Since this is a new feature, we need to update our tests to reflect the new functionality. You can find related tests in tests/unit/test_events.py or integration/test_pr_comment.py

**Comment by IceWreck** [comment]
@TomasTomecek 
I was trying to update/add test cases and it turns out the _unit/test_events.py_ and _integration/test_pr_comment.py_ tests already pass because the existing tests are just a subset of what we are doing now.

The actual issue is in _integration/test_release_event.py_

The following tests fail:

- tests/integration/test_release_event.py::test_dist_git_push_release_handle_multiple_branches FAILED [ 97%]
- tests/integration/test_release_event.py::test_dist_git_push_release_handle_one_failed FAILED [ 98%]
- tests/integration/test_release_event.py::test_dist_git_push_release_handle_all_failed FAILED [100%] 

However the below mentioned test, which is almost identical to the above tests passes.

- tests/integration/test_release_event.py::test_dist_git_push_release_handle PASSED [ 96%]


I haven't figured out the cause of the issue and do not understand why making new github releases would interfere with this.
Can you please take a look and tell me what I'm missing?

**Comment by jpopelka** [comment]
Looks like the current `master` is failing with the same errors, so it's not your fault.

**Comment by lachmanfrantisek** [comment]
@IceWreck Try rebasing your PR, please.

**Comment by IceWreck** [comment]
@lachmanfrantisek its already rebased. Did it yesterday. Its not behind compared to the main repo.

**Comment by TomasTomecek** [comment]
@IceWreck but the check says it's not rebased: https://softwarefactory-project.io/logs/33/433/cbc32a42577829985a349a810958d8f8a01f347e/check/packit-service-pre-commit/a602c31/log-classify.html

and also https://github.com/packit-service/packit-service/pull/431 got merged 2 minutes ago, so this definitely needs a rebase

**Comment by lachmanfrantisek** [comment]
@IceWreck 

Your branch starts on the merge commit of the pr #429 :

Screenshot from 2020-02-28 10-35-36

**Comment by TomasTomecek** [comment]
I want my PR #441 to be merged ASAP but can wait for your to be merged first, so please rebase

**Comment by IceWreck** [comment]
Okay, doing it rn.

**Comment by TomasTomecek** [comment]
uhhh, had to rebase as well after @lachmanfrantisek's got merged, so let's wait another 20 minutes

**Comment by IceWreck** [comment]
Rebased and squashed, but its still failing pre commit.

**Comment by csomh** [comment]
> Rebased and squashed.

You can see why this is still failing here:

https://softwarefactory-project.io/zuul/t/local/build/1559f7f610e1461fada0d33e51877fb4/log/job-output.txt#211

We use `black` to format the code, and black will fail if it has to re-format your code. Which is OK.

The only thing you have to do locally, is to amend your commit with the changes done by black and push again.

**Comment by IceWreck** [comment]
@csomh 

Thanks. Done.

```
files were modified by this hook

reformatted /home/zuul-worker/src/github.com/packit-service/packit-service/packit_service/worker/jobs.py
1 file reformatted, 79 files left unchanged.
```

This part threw me off and I thought it automatically reformatted itself.

**Comment by TomasTomecek** [comment]
Thanks for bearing with our comments!

> @TomasTomecek
> I was trying to update/add test cases and it turns out the _unit/test_events.py_ and _integration/test_pr_comment.py_ tests already pass because the existing tests are just a subset of what we are planning to do now.

That's not how testing works really - when you code a new feature, you want to make sure it really works by creating at least a few test cases. Ideally those would be integration/functional to exercise the code as much as possible. I don't think any of the existing tests exercise the feature you just implemented.

In this case, it's small amount of code, so I'd say it should suffice to just create a few unit tests to make sure the command is parsed correctly, something like:
```python
@pytest.mark.parametrize(
  "comment",
  (
    "/packit build",
    "asd\n/packit build\n",
    "no packit build",
    ...
)
def test_packit_command_parsing(comment):
  e = TheEvent()
  e.comment = comment
  assert ... SteveJobs.process_job_comments(e)
```

**Comment by IceWreck** [comment]
@TomasTomecek Done!

**Comment by jpopelka** [comment]
@IceWreck do you want to take a look at #463 ?

**Comment by IceWreck** [comment]
Sure, will fix it tonight.

**Pull request merged**